src/App.tsx

```diff
@@ -17,18 +84,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                searchValue={searchValue}
+                selectValue={selectValue}
+                onSearchInput={handleSearchInput}
+                onSelectInput={handleSelectInput}
+                onClearSearch={handleClearSearchInput}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todoListLoaded
+                ? (
+                  <TodoList
+                    todos={visibleTodos}
+                    selectedTodoId={selectedTodo ? selectedTodo.id : null}
```

Comment:
make this field optional

```suggestion
                    selectedTodoId={selectedTodo?.id}
```